helix-core/src/main/java/org/apache/helix/controller/stages/ResourceValidationStage.java

```diff
@@ -41,6 +42,14 @@ public void process(ClusterEvent event) throws Exception {
     if (cache == null) {
       throw new StageException("Missing attributes in event:" + event + ". Requires DataCache");
     }
+
+    // Check if cluster is still in management mode. Eg. there exists any frozen live instance.
+    if (!cache.shouldRunManagementPipeline()) {
+      // Trigger an immediate management mode pipeline.
+      RebalanceUtil.scheduleManagementModePipeline(event.getClusterName());
```

Comment:
Question: What would happen if you schedule onDemand pipeline here? Sine it is paused, the pipeline would go to Management?

Reply:
Basically, my question was why did you introduce a new event which is similar to ondemand pipeline? Since you updated this PR and you are now using Ondemand event type, this can be resolved.